### PR TITLE
Unstructured data first drafts

### LINKED ISSUE #8482: Add to tutorial + recipes + docs page about building sites with "unstructured data"
A persistent criticism of Gatsby is that you _have_ to use source plugins + GraphQL to build sites e.g. https://twitter.com/thekitze/status/1043532445742784515

The criticism is that this can complicate things unnecessarily for smaller sites.

The criticism is 100% correct! But it's also incorrect as you can write the exact same code in the above tweet using the node `createPages` API.

We should make this more clear to give people trying out Gatsby and easier onboarding to using the product.

We should add a brief section to the tutorial about building a small site from data queried from a simple REST api e.g. https://jsonplaceholder.typicode.com/

Also a docs page + prominent mention in the recipes page.

We should also mention the tradeoffs to this approach e.g. no gatsby-image or other transforms + more difficult time w/ medium+ complex sites, etc.

/cc @shannonbux @amberleyromo  

Amberley:
- [x] initial stub (thanks Shannon!)
- [ ] blog post (PR: #9188)
- [ ] update docs page (drawing from blog content)
- [ ] add prominent mention in recipes page

### PR DESCRIPTION
**Pull request by shannonbux**
Partially resolves #8482

### CODE REVIEW

File: docs/docs/using-unstructured-data.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+Absolutely not! You can use the node `createPages` API to pull unstructured data into Gatsby sites rather than GraphQL and source plugins. This is a great choice for small sites, while GraphQL and source plugins can help save time with more complex sites.
+
+### Prerequisites (if any)
+
+### The facts
+
+### Example
+
+**to do** Possibly pull from these examples
+https://twitter.com/thekitze/status/1043532445742784515
```
Comment (amberleyromo) [inline docs/docs/using-unstructured-data.md]: This is an example of the type of observation we're trying to explain toward -- imo relevant for context in the issue, not to be in the doc, I don't think.

File: docs/docs/using-unstructured-data.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+### The Gatsby recommendation
+
+If you're building a small site, one efficient way to build it is to pull in unstructured data as outlined in this guide, using `createPages` API, and then if the site becomes more complex later on, you move on to building more complex sites, or you'd like to transform your data, follow these steps:
+
+1.  Check out the [Plugin Library](/packages/) to see if the source plugins and/or transformer plugins you'd like to use already exist
+2.  If they don't exist, read the [Plugin Authoring](/docs/plugin-authoring/) guide and consider building your own!
+
+### Other resources
+
+- Link to a blogpost
```
Comment (amberleyromo) [inline docs/docs/using-unstructured-data.md]: Intending to merge this listing of resource types?

File: docs/tutorial/part-four/index.md [hunk 3]
```diff
@@ -50,7 +50,22 @@ and APIs of all sorts.
 **Gatsby's data layer lets you pull data from these (and any other source)
 directly into your components**—in the shape and form you want.
 
-## How Gatsby's data layer uses GraphQL to pull data into components
+## Using Unstructured Data vs GraphQL
```
Comment (amberleyromo) [inline docs/tutorial/part-four/index.md]: I'm not sure "Using Unstructured Data" as a top-level heading is a good phrase in the tutorial. I'd say maybe nix this, and just leave the intro to this with "Do I have to use GraphQL and source plugins to pull data into Gatsby sites?"

The phrase "unstructured data" makes more sense after this question

**Comment by amberleyromo** [comment]
@shannonbux You're intending for this draft to close this issue out? Or do you want this to remain on my immediate to-do list to flesh out further?

**Comment by shannonbux** [comment]
@Amberley Romo <amberley@gatsbyjs.com> Yeah, if you have the time, that'd
be great if you could finish it out and edit it however you see fit! I
thought I'd just get a start and take it as far as I could, knowing that
someone else would have to finish it :)

**Comment by amberleyromo** [comment]
@shannonbux that's great! Could you merge this but leave the issue open, or make me a new issue to keep it on my radar? 😸

**Comment by shannonbux** [comment]
Sure!

**Comment by shannonbux** [comment]
@amberleyromo closed and disconnected from issue!

**Pull request merged**